PENG-2585: CI to validate image

zsinx6 opened the pull request:
https://app.clickup.com/t/18022949/PENG-2585

matheushent commented on .github/workflows/test-image.yaml:
```diff
@@ -0,0 +1,56 @@
+      - name: run image
+        run: |
+          bash democluster/helpers/deploy_local_democluster.sh
+      - name: valitate snaps
+        run: |
+          SNAP_LIST=$(multipass exec democluster -- snap list)
+          SNAPS=("vantage-agent" "jobbergate-agent")
+
+          for SNAP in "${SNAPS[@]}"; do
+            if echo "$SNAP_LIST" | grep -q "$SNAP"; then
+              echo "$SNAP is installed."
+            else
+              echo "$SNAP is NOT installed."
+              exit 1
+            fi
+          done
```
I think we should verify here if the agents are active. For example:
```bash
multipass exec democluster -- systemctl is-active snapd.vantage-agent.daemon
```
It should return `active`, otherwise it's not working properly

matheushent commented on .github/workflows/test-image.yaml:
```diff
@@ -0,0 +1,56 @@
+      - name: check slurm is active
+        run: |
+          multipass exec democluster -- systemctl is-active slurmd
+      - name: test srun
+        run: |
+          multipass exec democluster -- srun ls
+      - name: check influx-db
+        run: |
+          multipass exec democluster -- systemctl is-active influxdb
```
I think we can stress the demo cluster a little bit here so we can test if it's sending metrics to influxdb properly. For example:
```bash
cat << 'EOF' > ./script.sh
#!/bin/bash
#SBATCH --job-name=cpu_stress
#SBATCH --ntasks=1
#SBATCH --time=0:02:00

start_time=$(date +%s)
duration=60

while true; do
    for i in {1..1000}; do
        echo "scale=10; s($i)/c($i)" | bc -l >/dev/null 2>&1
    done
    current_time=$(date +%s)
    elapsed=$((current_time - start_time))
    if [ $elapsed -ge $duration ]; then
        break
    fi
done

echo "CPU stress test completed after $elapsed seconds"
EOF

multipass transfer ./script.sh democluster-tosta-democluster:/home/ubuntu
multipass exec democluster -- sbatch /home/ubuntu/script.sh

# for loop to watch while the job is executing
# check in sacct if the job is completed

multipass exec democluster -- influx -database 'slurm-job-metrics' -execute 'show measurements' -host 'localhost'
# check output

multipass exec democluster -- influx -database 'slurm-job-metrics' -execute 'select * from CPUUtilization' -host 'localhost'
# check output
```

matheushent reviewed (dismissed):
Nice work, however I think we can still improve a bit :)

matheushent committed:
feat: [PENG-2585] create CI validation for democluster images.

This commit creates a new workflow that runs on PR whose purpose is to validate
the democluster image. It is done by completing a suite of checks:
1. Build the image;
2. Run the deploy script;
3. Verify if the snaps are installed and active;
4. Verify if the `slurmd` daemon is active;
5. Run a simple job using the `srun` command;
6. Check if the `influxdb` daemon is active;
7. Run a simple job to stress the CPU and check if metrics are available in InfluxDB.
